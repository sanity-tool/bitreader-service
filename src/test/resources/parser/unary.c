int *I_P, I;

static void addressOf() {
    I_P = &I;
}

static void deref() {
    *I_P = I;
}