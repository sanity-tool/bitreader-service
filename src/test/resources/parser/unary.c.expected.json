{
  "functions" : [ {
    "ref" : {
      "kind" : "Global",
      "index" : 0
    },
    "blocks" : [ {
      "instructions" : [ {
        "kind" : "LLVMStore",
        "operands" : [ {
          "kind" : "Global",
          "index" : 1
        }, {
          "kind" : "Global",
          "index" : 2
        } ]
      }, {
        "kind" : "LLVMRet"
      } ]
    } ],
    "entryBlockIndex" : 0
  }, {
    "ref" : {
      "kind" : "Global",
      "index" : 3
    },
    "blocks" : [ {
      "instructions" : [ {
        "kind" : "LLVMLoad",
        "operands" : [ {
          "kind" : "Global",
          "index" : 1
        } ]
      }, {
        "kind" : "LLVMLoad",
        "operands" : [ {
          "kind" : "Global",
          "index" : 2
        } ]
      }, {
        "kind" : "LLVMStore",
        "operands" : [ {
          "kind" : "Instruction",
          "index" : 0
        }, {
          "kind" : "Instruction",
          "index" : 1
        } ]
      }, {
        "kind" : "LLVMRet"
      } ]
    } ],
    "entryBlockIndex" : 0
  } ],
  "types" : [ {
    "kind" : "LLVMPointerTypeKind",
    "types" : [ 1 ]
  }, {
    "kind" : "LLVMFunctionTypeKind",
    "types" : [ 2 ]
  }, {
    "kind" : "LLVMVoidTypeKind"
  }, {
    "kind" : "LLVMPointerTypeKind",
    "types" : [ 4 ]
  }, {
    "kind" : "LLVMIntegerTypeKind",
    "len" : 32
  }, {
    "kind" : "LLVMPointerTypeKind",
    "types" : [ 3 ]
  } ],
  "globals" : [ {
    "kind" : "LLVMFunctionValueKind",
    "name" : "addressOf",
    "typeId" : 0
  }, {
    "kind" : "LLVMGlobalVariableValueKind",
    "name" : "I",
    "typeId" : 3
  }, {
    "kind" : "LLVMGlobalVariableValueKind",
    "name" : "I_P",
    "typeId" : 5
  }, {
    "kind" : "LLVMFunctionValueKind",
    "name" : "deref",
    "typeId" : 0
  } ]
}